src/timeline-manager.tsx

```diff
@@ -1,70 +1,84 @@
 // @ts-ignore
 import {SegmentsWrapper} from './components/chapters';
-import {OnClickEvent} from '@playkit-js/common/dist/hoc/a11y-wrapper';
 import {getTimeInText} from './utils';
+
 // @ts-ignore
-const {preact, redux, reducers, style, components} = KalturaPlayer.ui;
-const {PLAYER_SIZE} = components;
-// @ts-ignore
+const {core, ui} = KalturaPlayer;
+const {EngineType} = core;
+const {preact, redux, reducers, style, components} = ui;
+const {PLAYER_SIZE, ProgressIndicator, SeekBarPreview} = components;
 const {actions} = reducers.seekbar;
-// @ts-ignore
-const {core} = KalturaPlayer;
+
 const KalturaPlayerSeekBarSelector = '.playkit-seek-bar';
-const quizQuestionType = 'QuizQuestion';
-const chapterType = 'Chapter';
 const chaptersClassName = 'playkit-chapters';
-const {EngineType} = core;
 
 /**
  * @class TimelineManager
  */
 class TimelineManager {
-  _uiManager: any;
   _store: any;
   _cuePointsRemoveMap: Map<string, Function>;
   _cuePointsMap: Map<number, TimelineMarkerDataObject>;
   _chapters: Chapter[] = [];
   _resolveTimelineDurationPromise = () => {};
   _timelineDurationPromise: Promise<void>;
+  _getThumbnailInfoFn: (virtualTime: number) => ThumbnailInfo | Array<ThumbnailInfo>;
 
   /**
    * @constructor
    * @param _player
    * @param _logger
-   * @param _dispatchTimelineEvent
    * @param _eventManager
    */
-  constructor(
-    private _player: KalturaPlayerTypes.Player,
-    private _logger: any,
-    private _dispatchTimelineEvent: (event: string, payload: any) => void,
-    private _eventManager: any
-  ) {
-    this._uiManager = this._player.ui;
+  constructor(private _player: KalturaPlayerTypes.Player, private _logger: any, private _eventManager: any) {
     this._store = redux.useStore();
     this._cuePointsRemoveMap = new Map();
     this._cuePointsMap = new Map();
-    this._player.ready().then(() => {
-      if (this.dualScreenPlugin) {
-        this._addSeekBarPreview();
-      }
-    });
     this._timelineDurationPromise = this._makeTimelineDurationPromise();
+    this._getThumbnailInfoFn = this._player.getThumbnail.bind(this._player);
   }
 
+  get _uiManager() {
+    return this._player.ui;
+  }
   get _state(): any {
     return this._store.getState();
   }
 
+  public get timelineManagerAPI() {
+    return {
+      loadMedia: this.loadMedia,
+      addKalturaCuePoint: this.addKalturaCuePoint,
+      removeAllKalturaCuePoints: this._removeAllCuePoints,
+      addCuePoint: this.addCuePoint,
+      removeCuePoint: this.removeCuePoint,
+      setSeekbarPreview: this.setSeekbarPreview,
+      setGetThumbnailInfo: (fn: (virtualTime: number) => ThumbnailInfo | Array<ThumbnailInfo>) => {
+        this._addSeekBarPreview();
+        this._getThumbnailInfoFn = fn;
+      },
+      addSeekBarPreview: this._addSeekBarPreview,
+      // Expose the timelineManager to Cypress for testing purposes
+      ...(window.Cypress ? {timelineManager: this} : {})
```

Comment:
I don't like the reference from logic to testing code. 
is it possible to do something like initialize the timeline mgr in dev mode so it would just expose the service on the window, instead of explicitly expecting a Cypress object to be on the window ?
let's talk about it.